tests/gbs/test_data.py

```diff
@@ -0,0 +1,131 @@
+    ``DATASETS_LIST`` variable above."""
+
+    @pytest.fixture
+    def dataset(self, datasets):
+        """Fixture for loading each of the datasets in ``DATASETS_LIST``"""
+        yield datasets()
+
+    patch_samples = [
+        [0, 0, 0, 0],
+        [0, 0, 0, 1],
+        [0, 0, 1, 0],
+        [0, 0, 1, 1],
+        [0, 1, 0, 0],
+        [0, 1, 0, 1],
+        [0, 1, 1, 0],
+        [0, 1, 1, 1],
+        [2, 1, 1, 1],
+        [3, 0, 1, 1],
+    ]
+
+    @pytest.fixture
+    def dataset_patched(self, monkeypatch, datasets):
+        """Fixture for loading each of the datasets in ``DATASETS_LIST`` with the ``__init__``
+        monkeypatched to load the above ``patch_samples``."""
+
+        def mock_init(_self):
+            """Replacement ``__init__`` for all the datasets in ``DATASETS_LIST``"""
+            # pylint: disable=protected-access
+            _self._data = scipy.sparse.csr_matrix(self.patch_samples)
+            _self.adj = np.ones((4, 4))
+            _self.n_samples, _self.modes = 10, 4
+
+        with monkeypatch.context() as m:
+            m.setattr(datasets, "__init__", mock_init)
+            yield datasets()
+
+    def test_filename(self, dataset):
+        """Test if filename is valid string for each dataset"""
+        # pylint: disable=protected-access
+        assert isinstance(dataset._data_filename, str)
+
+    def test_n_mean(self, dataset):
+        """Test if mean photon number is valid float or int for each dataset"""
+        assert isinstance(dataset.n_mean, (float, int))
+        assert dataset.n_mean >= 0
+
+    def test_n_max(self, dataset):
+        """Test if maximum photon number is valid float or int for each dataset"""
+        assert isinstance(dataset.n_max, (float, int))
+        assert dataset.n_max >= 0
+
+    def test_threshold(self, dataset):
+        """Test if threshold flag is valid bool for each dataset"""
+        assert isinstance(dataset.threshold, bool)
+
+    def test_counts_ax0(self, dataset_patched):
+        """Test if dataset ``counts`` method returns the number of counts in each mode when
+        ``axis==0``. The samples are patched to the fixed ``patch_samples`` list."""
+        assert dataset_patched.counts(0) == [5, 5, 6, 6]
+
+    def test_counts_ax1(self, dataset_patched):
+        """Test if dataset ``counts`` method returns the number of counts in each sample when
+        ``axis==1``. The samples are patched to the fixed ``patch_samples`` list."""
+        assert dataset_patched.counts(1) == [0, 1, 1, 2, 1, 2, 2, 3, 5, 5]
+
+    def test_iter(self, dataset_patched):
+        """Test if dataset class allows correct iteration over itself"""
+        assert [i for i in dataset_patched] == self.patch_samples
+
+    def test_slice(self, dataset_patched):
+        """Test if dataset class allows correct slicing over items"""
+        assert [i for i in dataset_patched[1, 4, 2]] == [

```

Comment:
Is it normal to spread `[self.patch_samples[1], self.patch_samples[3], ]` over so many lines?

Reply:
It was an automatic thing from `black` 🤔